src/core_io.h

```diff
@@ -53,8 +53,7 @@ UniValue ValueFromAmount(const CAmount amount);
 std::string FormatScript(const CScript& script);
 std::string EncodeHexTx(const CTransaction& tx, const int serializeFlags = 0);
 std::string SighashToStr(unsigned char sighash_type);
-void ScriptPubKeyToUniv(const CScript& scriptPubKey, UniValue& out, bool include_hex, bool include_address = true);
-void ScriptToUniv(const CScript& script, UniValue& out);
-void TxToUniv(const CTransaction& tx, const uint256& hashBlock, UniValue& entry, bool include_hex = true, int serialize_flags = 0, const CTxUndo* txundo = nullptr, TxVerbosity verbosity = TxVerbosity::SHOW_DETAILS);
+void ScriptToUniv(const CScript& script, UniValue& out, bool include_hex = true, bool include_address = false);
```

Comment:
I wonder why we don't return the UniValue (or `optional<UniValue>` in case it can fail) instead of having it as second parameter. It would seem better API design to me.

Reply:
I think this is a great suggestion. And at first glance it seems to be straightforward enough to include in this PR. I'll follow up shortly

Reply:
I am leaning towards not including this change in this PR. I do think it is a great opportunity to cleanup and improve the code quality, so would like to follow up with another PR doing this. This pattern is also used all over the place, not just `ScriptToUniv`, but `TxToUniv`, `entryToJSON`, and probably more. Anyways I want to spend some more time on this and don't want to end up doing too much in this PR

Reply:
Can you take a quick look at the last two commits in this draft PR #23507?

a) is this along the lines of what you are thinking?
b) do you think it is worthwhile for me to do this across the codebase in #23507, eg `entryToJSON`, `TxToJSON`, etc... can be improved in the same way

Feel free to answer those in the linked PR and not here